internal/build/build.go

```diff
@@ -18,17 +18,223 @@ limitations under the License.
+		KernelCmdline:        d.Installation.KernelCmdLine,
+	}
+
+	logger.Info("Preparing installation setup")
+	dep, err := action.DigestInstallSetup(system, installFlags)
+	if err != nil {
+		logger.Error("Preparing installation setup failed")
+		return err
+	}
+
+	boot, err := bootloader.New(dep.BootConfig.Bootloader, system)
+	if err != nil {
+		logger.Error("Parsing boot config failed")
+		return err
+	}
+
+	manager := firmware.NewEfiBootManager(system)
+	upgrader := upgrade.New(ctx, system, upgrade.WithBootManager(manager), upgrade.WithBootloader(boot))
+	installer := install.New(ctx, system, install.WithUpgrader(upgrader))
+
+	logger.Info("Installing OS")
+	if err = installer.Install(dep); err != nil {
+		logger.Error("Installation failed")
+		return err
+	}
+
+	logger.Info("Installation complete")
+
+	return nil
+}
+
+func resolveManifest(manifestURI, storeDir string) (*resolver.ResolvedManifest, error) {
+	manifestStore := filepath.Join(storeDir, "release-manifests")
+	if err := os.MkdirAll(manifestStore, 0700); err != nil {
+		return nil, fmt.Errorf("creating release manfiest store '%s': %w", manifestStore, err)
+	}
+
+	extr, err := extractor.New(extractor.WithStore(manifestStore))
+	if err != nil {
+		return nil, fmt.Errorf("initialising OCI release manifest extractor: %w", err)
+	}
+
+	res := resolver.New(source.NewReader(extr))
+	m, err := res.Resolve(manifestURI)
+	if err != nil {
+		return nil, fmt.Errorf("resolving manifest at uri '%s': %w", manifestURI, err)
+	}
+
+	return m, nil
+}
+
+func writeConfigScript(d *image.Definition, dest string) (string, error) {
+	const configScriptName = "config.sh"
+
+	values := struct {
+		Users []image.User
+	}{
+		Users: d.OperatingSystem.Users,
+	}
+
+	data, err := template.Parse(configScriptName, configScriptTpl, &values)
+	if err != nil {
+		return "", fmt.Errorf("parsing config script template: %w", err)
+	}
+
+	filename := filepath.Join(dest, configScriptName)
+	if err = os.WriteFile(filename, []byte(data), os.FileMode(0o744)); err != nil {
+		return "", fmt.Errorf("writing config script: %w", err)
+	}
+	return filename, nil
+}
+
+func createDisk(runner sys.Runner, img image.Image, diskSize image.DiskSize) error {
+	const defaultSize = "10G"
+
+	if diskSize == "" {
+		diskSize = defaultSize
+	} else if !diskSize.IsValid() {
+		return fmt.Errorf("invalid disk size definition '%s'", diskSize)
+	}
+
+	_, err := runner.Run("qemu-img", "create", "-f", "raw", img.OutputImageName, string(diskSize))
+	return err
+}
+
+func attachDevice(runner sys.Runner, img image.Image) (string, error) {
+	out, err := runner.Run("losetup", "-f", "--show", img.OutputImageName)
+	if err != nil {
+		return "", err
+	}
+
+	device := strings.TrimSpace(string(out))
+	return device, nil
+}
+
+func detachDevice(runner sys.Runner, device string) error {
+	_, err := runner.Run("losetup", "-d", device)
+	return err
+}
+
+func addRKE2ToOverlays(ctx context.Context, downloadURL, overlaysDir string) error {
```

Comment:
I don't think we should have any RKE2 specific logic in here, in fact, the method I don't think it does any RKE2 specific thing. Couldn't we simply rename it to `addExtensionToOverlay` or similar? We will also need to pull the the elemental3-toolkit extension and we could use this method.